cps/tasks/metadata_extract.py

```diff
@@ -115,16 +115,41 @@ def run(self, worker_thread):
                                 self.progress = 0
                         finally:
                             log.info("Shelf title: %s", self.shelf_title)
+                            response = requests.get(self.original_url, params={"current_user_name": self.current_user_name, "shelf_title": self.shelf_title})
+                            if response.status_code == 200:
+                                self.shelf_id = response.json()["shelf_id"]
+                            else:
+                                log.error("An error occurred while trying to send the shelf title to %s", self.original_url)
+                            
+                        # update the metadata of every video in the shelf
+                        for index, requested_url in enumerate(requested_urls.keys()):
+                            try:
+                                p = process_open([lb_executable, "tubeadd", requested_url], newlines=True)
+                                p.wait()
+                            except Exception as e:
+                                log.error("An error occurred during updating the metadata of %s: %s", requested_url, e)
+                                self.message = f"{requested_url} failed: {e}"
+                        for index, requested_url in enumerate(requested_urls.keys()):
+                            try:
+                                view_count = conn.execute("SELECT view_count FROM media WHERE path = ?", (requested_url,)).fetchone()[0]
+                                time_uploaded = conn.execute("SELECT time_uploaded FROM media WHERE path = ?", (requested_url,)).fetchone()[0]
+                                time_uploaded = datetime.utcfromtimestamp(time_uploaded)
+                                now = datetime.now()
+                                # calculate views per day
+                                days_since_publish = (now - time_uploaded).days
```

Comment:
Can we force `days_since_publish` to be 1 at minimum (i.e. 1 day) to avoid divide-by-zero errors?